mobile/src/main/java/org/openhab/habdroid/model/Widget.kt

```diff
@@ -243,6 +243,8 @@ data class Widget(
                 state.toParsedState(item.state?.asNumber?.format)
             else -> state.toParsedState()
         }
+
+        internal val stateLabelRegex = Regex("\\[(.*)\\]$")
```

Comment:
You could modify the regex to not match [ ] inside the [ ] pair. That would make sure you get the last pair for the pattern and the rest stays in the label. That assumes the pattern or transformation cannot contain [ or ].

Reply:
I'm not sure I understand. Are you suggesting something like `\\[[^\\[\\]]*\\]`? What if the "label" sent by the server is `Label [value that contains [] chars]` ?

Reply:
Indeed, that won’t work well either. There is always something.
FWIW in BasicUI, at initialization of the field, some analysis is done on the pattern (which is the content in the [ ]). Core has that available. I don’t remember if that is available in the Android app. It is not in the SSE events for sure. That allows avoiding the problem. The assumption is that there is just a formatting pattern in the square brackets, not a transformation. See the BasicUI fix PR that touches this code (https://github.com/openhab/openhab-webui/pull/2685). The java class is for the initialization, the js code works handles the updates and SSE events.
In core, a good starting point to get an idea of what is happening is in the getDisplayState method, creating the displayState for SSE.

Reply:
For the time being, we'll have to assume and expect that the "label" part must not contain `[]` characters. I had a look in core code, and it has the same assumption.